rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -41,14 +42,18 @@ void
 SequentialReader::open(
   const StorageOptions & storage_options, const ConverterOptions & converter_options)
 {
+  storage_options_ = storage_options;
   storage_ = storage_factory_->open_read_only(storage_options.uri, storage_options.storage_id);
   if (!storage_) {
     throw std::runtime_error("No storage could be initialized. Abort");
   }
-  auto topics = storage_->get_metadata().topics_with_message_count;
+  const auto bag_metadata = storage_->get_metadata();
+  auto topics = bag_metadata.topics_with_message_count;
   if (topics.empty()) {
```

Comment:
I know this is not part of your changes, but I would feel like there should be a warning or something saying that the bagfile is empty.